components/Details.js

```diff
@@ -0,0 +1,74 @@
+import { useState, useEffect } from 'react';
+
+import { transformDetailsToFields } from './../utils/transformData';
+import { FormInput } from './Form/FormInput';
+export const Details = ({ detailsDataCallback, defaultValue }) => {
+  const [detailFields, setDetailFields] = useState(
+    transformDetailsToFields(defaultValue) || [],
```

Comment:
Consider either adding this array [{fieldName, fieldContent}, ... ] directly to firebase